### PR TITLE
Refactor application edit section

### PR DESCRIPTION
**Pull request by NipuniBhagya**

### CODE REVIEW

File: modules/form/src/utils/validate.ts [hunk 1]
```diff
@@ -56,9 +56,16 @@ export const getDefaultValidation = (field: string, fieldType: string, value: an
...
-    value: any, meta: any, field: string, fieldType: string, validation: any, required: boolean
+    value: any, meta: any, field: string, required: boolean, fieldType?: string, validation?: any
 ) => {
 
+    debugger;
+    if (!meta.modified) {
+        return;
+    }
+
+    console.log("Validation value", value);
```
Comment (inthirakumaaran) [inline modules/form/src/utils/validate.ts]: Do we need this console log?

File: apps/console/src/features/applications/components/forms/general-details-form.tsx [hunk 2]
```diff
@@ -105,17 +103,17 @@ export const GeneralDetailsForm: FunctionComponent<GeneralDetailsFormPopsInterfa
      * @param values - Form values.
      * @return {any} Sanitized form values.
      */
-    const updateConfigurations = (values: Map<string, FormValue>): any => {
-        return {
-            accessUrl: values.get("accessUrl").toString(),
+    const updateConfigurations = (values) => {
```
Comment (brionmario) [inline apps/console/src/features/applications/components/forms/general-details-form.tsx:106]:
```suggestion
    const updateConfigurations = (values): void => {
```

File: modules/form/src/components/field-button.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+export interface FieldButtonPropsInterface extends FormFieldPropsInterface {
+    /**
+     * Type of the button field.
+     */
+    buttonType: "primary_btn" | "cancel_btn" | "danger_btn" | "secondary_btn";
+}
+
+/**
+ * Implementation of the Button Field component.
+ * @param props
```
Comment (brionmario) [inline modules/form/src/components/field-button.tsx:34]: Missing return doc comment.

File: modules/form/src/components/adapters.tsx [hunk 4]
```diff
@@ -44,86 +44,132 @@ export const TextFieldAdapter = (props): ReactElement => {
...
             control={ Input }
+            autoFocus={ childFieldProps.autoFocus || false }
+            value={ meta.modified ? input.value : (childFieldProps?.value ? childFieldProps?.value : "") }
+            { ...omit(childFieldProps, ["value"]) }
+            error={ meta?.modified && meta?.error !== "" ? meta?.error : null }
+        />
+    );
+};
+
+export const PasswordFieldAdapter = (props): ReactElement => {
```
Comment (brionmario) [inline modules/form/src/components/adapters.tsx:59]: Shall we add interfaces for these adapters?

**Review by brionmario — APPROVED** [review]

**Review by DimalChandrasiri — APPROVED** [review]

**Pull request merged**